### PR TITLE
Retry for transient exceptions while doing cleanup for Hadoop Jobs

### PR DESCRIPTION
**Pull request by nishantmonu51**
fixes https://github.com/druid-io/druid/issues/1828

### CODE REVIEW

**Comment by fjy** [comment]
👍

**Comment by nishantmonu51** [comment]
travis failure with AnnouncerTest - https://travis-ci.org/druid-io/druid/jobs/139633994

File: indexing-hadoop/src/main/java/io/druid/indexer/JobHelper.java [hunk 1]
```diff
@@ -44,6 +44,7 @@
 import org.apache.hadoop.fs.Path;
 import org.apache.hadoop.io.retry.RetryPolicies;
 import org.apache.hadoop.io.retry.RetryProxy;
+import org.apache.hadoop.ipc.Server;
```
Comment (drcrallen) [inline indexing-hadoop/src/main/java/io/druid/indexer/JobHelper.java]: Unused import?

Comment (nishantmonu51) [inline indexing-hadoop/src/main/java/io/druid/indexer/JobHelper.java]: fixed.

**Comment by drcrallen** [comment]
@nishantmonu51 can you make the title describe what the patch does?

Also is there any test which would be appropriate to add for this?

**Comment by fjy** [comment]
@drcrallen I think difficult to test

File: indexing-hadoop/src/main/java/io/druid/indexer/updater/HadoopConverterJob.java [hunk 2]
```diff
@@ -147,8 +147,8 @@ public static void cleanup(Job job) throws IOException
   {
     final Path jobDir = getJobPath(job.getJobID(), job.getWorkingDirectory());
     final FileSystem fs = jobDir.getFileSystem(job.getConfiguration());
-    fs.delete(jobDir, true);
-    fs.delete(getJobClassPathDir(job.getJobName(), job.getWorkingDirectory()), true);
+    JobHelper.deleteWithRetry(fs, jobDir, true);
+    JobHelper.deleteWithRetry(fs, getJobClassPathDir(job.getJobName(), job.getWorkingDirectory()), true);
```
Comment (drcrallen) [inline indexing-hadoop/src/main/java/io/druid/indexer/updater/HadoopConverterJob.java]: Should this fire regardless of if the prior delete succeeds?

Comment (nishantmonu51) [inline indexing-hadoop/src/main/java/io/druid/indexer/updater/HadoopConverterJob.java]: yeah, fixed.

**Comment by drcrallen** [comment]
👍 after travis

**Pull request merged**